### PR TITLE
Readme and MFG sample scripts

### PR DESCRIPTION
**Pull request by EthanChangAED**
- Manufacturing sample script
- Readme.md for the sample script
- Wi-Fi SoftAP access info tool

### CODE REVIEW

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+| 1.0.201112.1001 | November 13. 2020 | (Open source version)<br>-Rename WifiDPP_PSK to Wifi-SoftAP_PSK. |
+
+## Introduction
+**Manufacturing Provisioning Tool Sample Script for Santa Cruz devices** is a Linux shell script for Santa Cruz device manufacturers to <ins>capture the device identities or hardware keys in the manufacturing line</ins>. The tool will generate a “device record” with those hardware identities, output it directly or save to a JSON file.
+The “device record” can then be used in post manufacturing process to provision each device to Azure as well as enabling Wi-Fi Zero Touch Provisioning scenario. However, **this tool only covers “collecting” the information for provisioning and does not handle the actual enrollment process to Azure**.
+
+## Hardware requirement and Prerequisite
+This script currently supports the following hardware. With minor customization (if necessary), it shall be able to support Santa Cruz device that loaded with Mariner OS.
+ - PE100 (i.MX 8)
+ - DKSC-101 (i.MX 8)
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:19]: What device is DKSC-101?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+The tool must be executed with the following environment requirements:
+
+ - Microsoft Mariner OS
+	 - (With other dependences below fulfilled, this tool should be able to run on various of Linux distros, or just need minor modifications.)
+ - TPM must be enabled
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:25]: How do device manufacturers enable this?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+ - TPM tool package ([tpm2-tools v4.2](https://github.com/tpm2-software/tpm2-tools/wiki)) must be included
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:26]: This has to be installed onto the device, not the PC that will be used to run the script?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        (pe101 is for DKSC-101)
+    -n, --serialnum=<serial_number> (Optional)
+        Given serial number to be compared with what've read from hardware.
+    -o, --overridesn (Optional)
+        Override device record with given serial number instead of serial number read from hardware.
+        (This approach is not recommended, but is an alternative when S/N is not yet store in the firmware.)
+    -f, --file=<output_filename> (Optional)
+        Redirect device record output from standard output to a file at /tmp.
+    --inittpm (Optional)
+        Clear TPM before to provision TPM.
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:49]: What exactly does it mean to clear the TPM before provisioning the TPM?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Other options:
+
+    -h, --help
+	    Apply this option solely to show help messages.
+    -c, --checkrequired
+	    Apply this option with "-m=<model_name>" to run system requirement check and then exit.
+    -q, --query
+	    Apply this option with "-m=<model_name>" to run this tool in query mode (dry-run) without re-provisioning TPM.
+**Executing and example**
+To execute this tool. Copy the script file (aed-mfgtool-azuredeviceprovision.devkit.sh) to target device (PE100/DKSC-101) through SSH connection:
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:60]: Will they connect to the device's SoftAP first, then connect over SSH through putty?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    "Wifi_MAC": "0A:0B:03:04:05:06",
+    "Wifi-SoftAP_PSK": "sdf234wh"
+    }
+    rc=0
+
+## Description of Outputs Data
+The following device identity/information (if applicable to the hardware) shall exists before performing the capturing with the script.
+
+ - SerialNumber
+	 - This shall be the key for associate the other device identities with the actual hardware. (Microsoft strongly recommend that the the S/N should be visible on the device. That way, there is an easy way to associate physical device with its virtual copy on the Azure side.)
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:94]: How so? Other device identities meaning the other outputs listed below?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+| Version | Released Date | Description |
+|--|--|--|
+| 0.1.200417.1 | April 24, 2020 | Beta version released. |
+| 0.2.200501.1 | May 20, 2020 | -Support PE101<br>-Provision/capture of Wifi ZTP key enabled. |
+| 0.3.200706.1 | July 10, 2020 | -Support read S/N on Mariner OS and checking mechanism.<br>-Modified/Add new fields: "WifiDPP_MAC, "WifiDPP_PSK". |
+| 0.4.200806.1 | August 15. 2020 | Minor bug fix. |
+| 1.0.201112.1001 | November 13. 2020 | (Open source version)<br>-Rename WifiDPP_PSK to Wifi-SoftAP_PSK. |
+
+## Introduction
+**Manufacturing Provisioning Tool Sample Script for Santa Cruz devices** is a Linux shell script for Santa Cruz device manufacturers to <ins>capture the device identities or hardware keys in the manufacturing line</ins>. The tool will generate a “device record” with those hardware identities, output it directly or save to a JSON file.
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:13]: But also for private preview users who need to find the SoftAP password, right?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	 - Used for enabling the Wifi ZTP (Zero Touch Provisioning) feature. Can also serve as a secondary device identity if necessary.
+**[Note]** WifiDPP_EccPub and Wifi_MAC are only required if the manufacturer want to enable the Wi-Fi ZTP feature and Wi-Fi module is included in the hardware design. However, pre-record the WifiDPP_EccPub at manufacturing would make the experience of future hardware/function expansion more seamless. (If there is a chance that SI/customer will install the Wi-Fi module and enable ZTP later, they would already have the ECC Key.)
+ - Wifi-SoftAP_PSK
+	 - This is also a derivative ID from the TPM key. Default password of the Wi-Fi SoftAP.
+
+## Known Issue
+
+ - [Fixed] Support for DKSC-101 to be enabled. 
+ - [Fixed] Wi-Fi ZTP key   provisioning and capturing is not implemented yet. Will be providing in future version of script. 
+ - [Won’t fix] For some early preview hardware, the tool cannot retrieve the completed S/N (last 4 digits will be “0000”). This is caused by a hardware issue, not an issue of
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:114]: So hardware distributed after a certain date won't have this issue? If so, what is the date?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    aed-mfgtool-azuredeviceprovision.devkit.sh [-m|--model=<model_name>] [-n|--serialnum=<serial_number>] [--inittpm]
+    [-h|--help] [-f|--file=<output_filename>] [-c|--checkrequired] [-o|--overridesn] [-q|--query]
+**Parameters**    
+Manufacturing featured options:
+
+    -m, --model=<model_name> (Required)
+        Name of supported models: pe100 pe101.
+        (pe101 is for DKSC-101)
+    -n, --serialnum=<serial_number> (Optional)
+        Given serial number to be compared with what've read from hardware.
```
Comment (elqu20) [inline tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:42]: I'm a little confused here. So this compares the device serial number to the serial number that you enter in the command? But it doesn't return true/false, as the example below shows. Which number does it return?

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md (hunk 1 above)
Comment (EthanChangAED) [reply tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:19]: DKSC-101 is the temp official name of devkit. It's also what got printed on the back side of the devkit.

File: tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md (hunk 2 above)
Comment (EthanChangAED) [reply tools/Manufacturing-Provisioning-Tool-Sample-Script/README.md:25]: Basically two key works,
1. The TPM chip needs to be correctly provisioned.
2. Driver and device tree need to be correctly integrated to the boot loader and OS.
I felt it's within the scope of platform bring up. But do you think it's better to add some more description here?

**Pull request merged**